torch/nn/quantized/_reference/modules/utils.py

```diff
@@ -16,7 +16,7 @@ def _init_weight_qparams(self, weight_qparams, device):
             None, torch.per_tensor_affine, torch.per_channel_affine,
             torch.per_channel_affine_float_qparams], \
             Exception(f"qscheme: {self.weight_qscheme} is not support in reference quantized {self._get_name()}")
-        if self.weight_dtype in [torch.quint8, torch.qint8, torch.quint4x2]:
+        if self.weight_dtype in [torch.quint8, torch.qint8, torch.quint4x2, torch.qint32]:
```

Comment:
can we add a warning on L37 in the else branch to enable the user to find this easier next time?

Reply:
I think we can add an assert, the `else` branch should not be called

Reply:
I added 'if-else' back due to one of fx test cases needs torch.float, so it will call 'else'

Reply:
TestQuantizeFx.test_dynamic_with_fusion_multiple_uses